dev/benchmarks/macrobenchmarks/test/util.dart

```diff
@@ -0,0 +1,192 @@
+│  builds. Benchmarks should be run using this command  ┆
+│  line:  flutter drive --profile test_perf.dart        ┊
+│                                                       ┊
+└─────────────────────────────────────────────────╌┄┈  🐢
+''';
+
+void macroPerfTest(
+  String testName,
+  String routeName, {
+  Duration pageDelay,
+  Duration duration = const Duration(seconds: 3),
+  Duration timeout = const Duration(seconds: 30),
+  Future<void> driverOps(WidgetTester tester),
+  Future<void> setupOps(WidgetTester tester),
+}) {
+  assert(() {
+    debugPrint(_kDebugWarning);
+    return true;
+  }());
+  final WidgetsBinding _binding = E2EWidgetsFlutterBinding.ensureInitialized();
+  assert(_binding is E2EWidgetsFlutterBinding);
+  final E2EWidgetsFlutterBinding binding = _binding as E2EWidgetsFlutterBinding;
+  binding.framePolicy = LiveTestWidgetsFlutterBindingFramePolicy.fullyLive;
+
+  testWidgets(testName, (WidgetTester tester) async {
+    assert((tester.binding as LiveTestWidgetsFlutterBinding).framePolicy ==
+        LiveTestWidgetsFlutterBindingFramePolicy.fullyLive);
+    app.main();
+    await tester.pumpAndSettle();
+
+    // The slight initial delay avoids starting the timing during a
+    // period of increased load on the device. Without this delay, the
+    // benchmark has greater noise.
+    // See: https://github.com/flutter/flutter/issues/19434
+    await tester.binding.delayed(const Duration(microseconds: 250));
+
+    final Finder scrollable =
+        find.byKey(const ValueKey<String>(kScrollableName));
+    expect(scrollable, findsOneWidget);
+    final Finder button =
+        find.byKey(ValueKey<String>(routeName), skipOffstage: false);
+    await tester.ensureVisible(button);
+    expect(button, findsOneWidget);
+    await tester.pumpAndSettle();
+    await tester.tap(button);
+
+    if (pageDelay != null) {
+      // Wait for the page to load
+      await tester.binding.delayed(pageDelay);
+    }
+
+    if (setupOps != null) {
+      await setupOps(tester);
+    }
+
+    await watchPerformance(binding, () async {
+      final Future<void> durationFuture = tester.binding.delayed(duration);
+      if (driverOps != null) {
+        await driverOps(tester);
+      }
+      await durationFuture;
+    });
+  }, semanticsEnabled: false, timeout: Timeout(timeout));
+}
+
+Future<void> watchPerformance(
+  E2EWidgetsFlutterBinding binding,
+  Future<void> action(),
+) async {
+  // This method might be good as part of e2e,
+  // so is the helper `PerformanceWatcher` class.
+  final PerformanceWatcher frameTimes = PerformanceWatcher();
+  final TimingsCallback watcher = frameTimes.addData;
+  binding.addTimingsCallback(watcher);
+  await action();
+  binding.removeTimingsCallback(watcher);
+  // TODO(CareF): determine if it's running on firebase and report metric online
+  binding.reportData = <String, dynamic>{'performance': frameTimes.summary};
+}
+
+class PerformanceWatcher {
```

Comment:
Nit: `FrameTimingSummarizer` seems to be a more informative name.

Reply:
Done